src/main/java/com/terracottatech/frs/compaction/CompactorImpl.java

```diff
@@ -254,14 +258,22 @@ private void compact() throws ExecutionException, InterruptedException {
     }
   }
 
+  @Override
+  public CompactionAction convertToAction(
+      ObjectManagerEntry<ByteBuffer, ByteBuffer, ByteBuffer> entry) {
+    PutAction delegate = new PutAction(objectManager, null,
+        entry.getId(), entry.getKey(), entry.getValue(), entry.getLsn());
+    return new CompactionAction(objectManager, entry, delegate);
+  }
+
```

Comment:
Why not just inject the two options as a factory in the constructor of CompactorImpl so you don't need to subclass.

Reply:
Yes, injecting a factory (for creation of CompactionAction) to CompactorImpl seems a better approach and it won't interfere the existing Compactor interface.